types/codemirror/addon/hint/show-hint.d.ts

```diff
@@ -61,8 +61,8 @@ declare module "codemirror" {
         closeOnUnfocus?: boolean;
         completeOnSingleClick?: boolean;
         container?: HTMLElement | null;
-        customKeys?: {[x: string]: () => void} | null;
-        extraKeys?: {[x: string]: () => void} | null;
+        customKeys?: {[key: string]: (editor: Editor, handle: Handle) => void} | null;
+        extraKeys?: {[key: string]: (editor: Editor, handle: Handle) => void} | null;
```

Comment:
Thanks for refinement the type 🙏 
I wonder whether the `extraKeys` properties can have `string` or `false` values.
And I wonder whether the functions can have the second argument.
Do you have any thoughts about these?

I found [the document](https://codemirror.net/doc/manual.html#option_extraKeys) says that `extraKeys` should be `a valid key map value`.
And this document says `The values of properties in key maps can be either functions of a single argument (the CodeMirror instance), strings, or false`.
Thus, I think we shold define `extraKeys` types as following:
```typescript
extraKeys?: { [key: string]: (editor: Editor) => void | string | false | null }
```
Is it correct? 🤔

Reply:
@azoson , Thanks for reviewing my PR.
I think [the document](https://codemirror.net/doc/manual.html#option_extraKeys) is describing the `extraKeys` for `codemirror` instance itself. The `extraKeys` we are refining is for `hint` addon of `codemirror`. Please refer to [this document](https://codemirror.net/doc/manual.html#addon_show-hint). It has its own `extraKeys` option.

> customKeys: keymap
> Allows you to provide a custom key map of keys to be active when the pop-up is active. The handlers will be called with an extra argument, a handle to the completion menu, which has moveFocus(n), setFocus(n), pick(), and close() methods (see the source for details), that can be used to change the focused element, pick the current element or close the menu. Additionally menuSize() can give you access to the size of the current dropdown menu, length give you the number of available completions, and data give you full access to the completion returned by the hinting function.
> extraKeys: keymap
> Like customKeys above, but the bindings will be added to the set of default bindings, instead of replacing them.

And its implementation is found [here](https://github.com/codemirror/CodeMirror/blob/5f722737883bbeadb2a629857d3961490e4134ca/addon/hint/show-hint.js#L183). 
I hope this helps you to understand

Reply:
@azoson , Thanks for reviewing my PR.
I think [the document](https://codemirror.net/doc/manual.html#option_extraKeys) is describing the `extraKeys` for `codemirror` instance itself. The `extraKeys` we are refining is for `hint` addon of `codemirror`. Please refer to [this document](https://codemirror.net/doc/manual.html#addon_show-hint). It has its own `extraKeys` option.

> customKeys: keymap
> Allows you to provide a custom key map of keys to be active when the pop-up is active. The handlers will be called with an extra argument, a handle to the completion menu, which has moveFocus(n), setFocus(n), pick(), and close() methods (see the source for details), that can be used to change the focused element, pick the current element or close the menu. Additionally menuSize() can give you access to the size of the current dropdown menu, length give you the number of available completions, and data give you full access to the completion returned by the hinting function.
> extraKeys: keymap
> Like customKeys above, but the bindings will be added to the set of default bindings, instead of replacing them.

And its implementation is found [here](https://github.com/codemirror/CodeMirror/blob/5f722737883bbeadb2a629857d3961490e4134ca/addon/hint/show-hint.js#L183). 
I hope this helps you to understand

Reply:
@azoson , Thanks for reviewing my PR.
I think [the document](https://codemirror.net/doc/manual.html#option_extraKeys) is describing the `extraKeys` for `codemirror` instance itself. The `extraKeys` we are refining is for `hint` addon of `codemirror`. Please refer to [this document](https://codemirror.net/doc/manual.html#addon_show-hint). It has its own `extraKeys` option.

> customKeys: keymap
> Allows you to provide a custom key map of keys to be active when the pop-up is active. The handlers will be called with an extra argument, a handle to the completion menu, which has moveFocus(n), setFocus(n), pick(), and close() methods (see the source for details), that can be used to change the focused element, pick the current element or close the menu. Additionally menuSize() can give you access to the size of the current dropdown menu, length give you the number of available completions, and data give you full access to the completion returned by the hinting function.
> extraKeys: keymap
> Like customKeys above, but the bindings will be added to the set of default bindings, instead of replacing them.

And its implementation is found [here](https://github.com/codemirror/CodeMirror/blob/5f722737883bbeadb2a629857d3961490e4134ca/addon/hint/show-hint.js#L183). 
I hope this helps you to understand